blog/_posts/2017-01-24-cudanative-nvidia.md

```diff
@@ -0,0 +1,234 @@
+---
+layout: post
+title:  GPU development in the Julia programming language
+author: <a href="https://github.com/maleadt">Tim Besard</a>
+cudanative_tree: https://github.com/JuliaGPU/CUDAnative.jl/blob/6e6ec23ef4def367dec6ae0ba3e43f1d9daae7ec
+---
+
+[Julia](http://julialang.org/) is a dynamic programming language for technical computing,
+with syntax that is familiar to users of other technical computing environments. The
+language has been created with performance in mind, and combines careful language design
+with a sophisticated LLVM-based compiler.
+
+Julia is well regarded for programming multicore CPUs and large parallel computing systems,
+but the tools for using Julia with GPUs are worth a look as well. In this blogpost, I will
+list some of those packages but focus on the latest addition: native GPU programming with
+CUDAnative.jl
+
+
+
+## Available tools
+
+The Julia package ecosystem contains quite a few GPU-related packages, targeting different
+levels of abstraction. At the highest abstraction level, domain-specific packages like
+[MXNet.jl](https://github.com/dmlc/MXNet.jl) or
+[TensorFlow.jl](https://github.com/malmaud/TensorFlow.jl) can transparently use any
+available GPUs. More generic development is possible with
+[ArrayFire.jl](https://github.com/JuliaComputing/ArrayFire.jl), and if you need a
+specialized implementation you can use wrappers for vendor-specific library packages
+(cuBLAS, cuFFT, cuSPARSE, ...) available at the [JuliaGPU GitHub
+organization](https://github.com/JuliaGPU/). Each of those packages is just a single
+`Pkg.add(...)` away!
+
+What's missing from this list is the lowest abstraction level, where you can create kernels
```

Comment:
`write kernels` instead of `create`? to emphasis that we are not loading C-kernels.